INSTALL.md

```diff
@@ -77,3 +77,14 @@ Please see [INSTALL.make.md](dev/doc/INSTALL.make.md) for build and
 installation instructions using `make`. If you wish to experiment with
 the Dune-based system see the [dune guide for
 developers](dev/doc/build-system.dune.md).
+
+Notes on build-time dependencies
+--------------------------------
+
+- The OCaml compiler and findlib are build-time dependencies, but also
+  run-time dependencies if you wish to use the native compiler.
+
+- When loading plugins, you should make sure that these were compiled
+  with the same version of OCaml as Coq.  This is also true when
+  loading `vo` files.  For this reason, it is rarely a good idea to
+  distribute pre-compiled `vo` files.
```

Comment:
It would be great to point out a way to determine which version of OCaml was used to compile a given .vo file.

Reply:
cc @ppedrot 

I don't think that's possible in fact. Even votour from Coq 8.11.1 fails on a vo file compiled with current master (and the converse is also true).

Reply:
We could store that information in the vo metadata. Anyways, there was recently a path by @ejgallego that stored the hash of the OCaml plugin files inside the vo, so that currently trying to load a vo with plugin dependencies mismatch would already trigger an error.

Reply:
See #12244.

Reply:
>This is also true when loading `vo` files

Is it really?

Reply:
>currently trying to load a vo with plugin dependencies mismatch would already trigger an error

That's not true, the hash is stored but not checked IIRC